### ISSUE TITLE
[Feature] command completion for zsh

### ISSUE THREAD

**Issue by santosh** [nerdvegas/rez#649]
As soon as rez installation is finished, the following message is shown:

```
You may also want to source the completion script (for bash):
source /home/sntshk/rez/completion/complete.sh
```

That directory contains 2 files, for bash and for csh. 

I'm not quite sure how autocompletion is handled in zsh _(autocompletion is the selling point of zsh)_. But there must be some way we can autocomplete package name after rez env?

Note: This issue is also related to #451.

**Comment by maxnbk** [comment]
I have a WIP zsh plugin attempt I plan to PR soon.

**Comment by nerdvegas** [comment]
Fixed by #711
